src/controllers/shopping-cart.controller.ts

```diff
@@ -78,6 +84,64 @@ export class ShoppingCartController {
     @param.path.string('userId') userId: string,
     @requestBody({description: 'shopping cart item'}) item: ShoppingCartItem,
   ) {
-    await this.shoppingCartRepository.addItem(userId, item);
+    debug('Add item %j to shopping cart %s', item, userId);
+    return retry(
+      {
+        run: () => this.shoppingCartRepository.addItem(userId, item),
+        description: `update the shopping cart for '${userId}'`,
+      },
+      10,
+      100,
+    );
   }
 }
+
+/**
+ * Retry a task for number of times with the given interval in ms
+ * @param task Task object {run, description}
+ * @param maxRetries Maximum number of tries, default to 10
+ * @param interval Milliseconds to wait after each try, default to 100ms
+ */
+async function retry<T>(
```

Comment:
do we want tests for this function?

Reply:
We could add a unit test but not sure how to test it with redis.

Reply:
Added tests.